### ISSUE TITLE
Warning about a missing 'default’ label a switch statement

### ISSUE THREAD

**Issue by richardrom** [llvm/llvm-project#90246]
Hi, 

I’ve recently been testing Clang 18.x.x and notice that it produces a warning about a missing default label when using 
enum class in switch statements when the _-Weverything_ flag is used. This was warning not produce using Clang 17. 
Weirdly enough, when you add the default label it now produces the warning _-Wcovered-switch-default_

The code can be easily seeing in compiler explorer.
This snippet is without using the default label
https://compiler-explorer.com/z/Mobxq3nzY
Using the default label:
https://compiler-explorer.com/z/Prde4zYrK

If in need of code:

```c++
enum class Enum {
    e1, e2, e3, e4
};


int test(Enum num) {
    switch(num) {
        case Enum::e1: return 0;
        case Enum::e2: return 1;
        case Enum::e3: return 2;
        case Enum::e4: return 3;
     //   default:;
    }
}
```
Compiler flags: -std=c++20 -Weverything -Wno-c++98-compat -Wno-missing-prototypes

Have I missed something in the C++ Standard??

**Comment by EugeneZelenko** [comment]
Duplicate of #81354.
